delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/model/ResourcesSearchParams.scala

```diff
@@ -57,6 +58,22 @@ final case class ResourcesSearchParams(
 
 object ResourcesSearchParams {
 
+  sealed trait TypeOperator
+
+  object TypeOperator {
+    case object AND extends TypeOperator
+    case object OR  extends TypeOperator
+
+    implicit val typeFromStringUnmarshaller: FromStringUnmarshaller[TypeOperator] =
+      Unmarshaller.strict[String, TypeOperator] { str =>
+        str.toLowerCase() match {
+          case "and" => TypeOperator.AND
+          case "or"  => TypeOperator.OR
+          case other => throw new IllegalArgumentException(s"'$other' is not a valid type operator (and/or)")
+        }
```

Comment:
using toLowerCase to just provide some slight ease-of-use, it should not cause us any problems

Reply:
Exception seems to be the way we are signalling errors in other `FromStringUnmarshaller` instances

Reply:
Yep, I don't think we can do it otherwise.
I think most of them live in `QueryParamsUnmarshalling` though.
So I would argue to move this there

Reply:
We will need to get rid of akka soon, so it is a good idea to avoid to mix it with classes that are not directly linked to it

Reply:
Arguably the TypeOperator could be placed in the sdk